db/migrate/20221130150352_create_project_journals.rb

```diff
@@ -0,0 +1,15 @@
+class CreateProjectJournals < ActiveRecord::Migration[7.0]
+  # rubocop:disable Rails/CreateTableWithTimestamps(RuboCop)
+  def change
+    create_table :project_journals do |t|
+      t.string :name, default: '', null: false
+      t.text :description
+      t.boolean :public, default: true, null: false
+      t.bigint :parent_id
+      t.string :identifier
+      t.boolean :active, default: true, null: false
+      t.boolean :templated, default: false, null: false
```

Comment:
For the constraint, if a constraint is later added, for instance on `description`, then the journal can not also have this constraint without rewriting history and setting a dummy value for older description values that were nil.

On the other hand, the point of having non-null constraints is to be sure that our code does not have to deal with `nil` values. Allowing `nil` values in journals defeats this as we would still have to nil-check everything. This will be especially true as baseline comparison will pull its data from journals.

For this reason I would keep the constraints.

For the default values, your reasoning sounds right, we should not need default values for journal tables.
If we should follow that route, I'd prefer doing this in a separate PR for all journal tables. I'll make an exception for `projects_journals.name` and `projects.name`.

Reply:
@ulferts so, should I issue a PR to remove defaults from all `xxx_journals` fields?

Reply:
@cbliard yet, but there is absolutely no hurry in doing so. You can also open a "Maintenance" ticket and we do it at a later time.

Reply:
https://community.openproject.org/wp/45234 created